x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -256,8 +274,16 @@ private void loadModel(String modelId, Consumer consumer) {
                 trainedModelCircuitBreaker.addEstimateBytesAndMaybeBreak(trainedModelConfig.getEstimatedHeapMemory(), modelId);
                 provider.getTrainedModelForInference(modelId, ActionListener.wrap(
                     inferenceDefinition -> {
-                        // Since we have used the previously stored estimate to help guard against OOM we need to adjust the memory
-                        // So that the memory this model uses in the circuit breaker is the most accurate estimate.
+                        try {
+                            // Since we have used the previously stored estimate to help guard against OOM we need
+                            // to adjust the memory so that the memory this model uses in the circuit breaker
+                            // is the most accurate estimate.
+                            updateCircuitBreakerEstimate(modelId, inferenceDefinition, trainedModelConfig);
+                        } catch (CircuitBreakingException ex) {
+                            handleLoadFailure(modelId, ex);
+                            return;
+                        }
+
                         long estimateDiff = inferenceDefinition.ramBytesUsed() - trainedModelConfig.getEstimatedHeapMemory();
```

Comment:
Do we need this with the call to `updateCircuitBreakerEstimate`

Reply:
🤦